src/Xamarin.Android.Build.Tasks/Tasks/GenerateResourceDesigner.cs

```diff
@@ -86,6 +86,10 @@ public override bool RunTask ()
 				var parser = new ManagedResourceParser () { Log = Log, JavaPlatformDirectory = javaPlatformDirectory, ResourceFlagFile = ResourceFlagFile };
 				resources = parser.Parse (ResourceDirectory, RTxtFile ?? string.Empty, AdditionalResourceDirectories?.Select (x => x.ItemSpec), IsApplication, resource_fixup);
 			} else {
+				if (JavaResgenInputFile == null) {
+					Log.LogCodedError ("XA1000", Properties.Resources.XA1000, nameof (JavaResgenInputFile));
+					return false;
```

Comment:
This error message, `XA1000`, doesn't match the problem!

Let's just change this to:
```suggestion
					throw new ArgumentNullException (nameof (JavaResgenInputFile));
```
And then the default error handling should log an appropriate message.